### PR TITLE
bt-368: Add step 5 preview screen onboarding

### LINKED ISSUE #368: Add Step 5 (Preview screen) to talent onboarding
Design link: [https://www.figma.com/file/Yd7USgP7PLvA0W794jHXWu/BSA_recruting?type=design&node-id=7318-9656&mode=design&t=Tm68KisNxezn0fDM-0](https://www.figma.com/file/Yd7USgP7PLvA0W794jHXWu/BSA_recruting?type=design&node-id=7318-9656&mode=design&t=Tm68KisNxezn0fDM-0)

User story:

As a Talent, I want to be able to see my profile, so that I can read/review all information that I filled in the previous steps

Acceptance criteria:

1. The step number “STEP 05/05” is placed in the upper right corner on the same line with the title of the page “Preview”
2. The content area of the page must contain:
- the information about the job title that the user chose
- the information about the size of salary expectation that the user chose
- the information about the years of experience that the user chose
- the information about the English level that the user chose
- the information about the employment type that the user chose
- the information about the vacancies that the user does not consider
- the information about the Talent's date account publishing
- the user’s cover letter
- active tabs: Scores & skills, Feedbacks, Project
- the active button “Save without publishing” 
- the active button “Publish now”

3. The tab “Scores & skills” must contain:
- the BSA badges with labels and scores
- the information about the skills that the user chose
- 
4. The tab “Feedbacks” must contain HR comments about the Talent's skills:

- information about personality type, the selected components are displayed as tags
- HR badges, the selected components are displayed as tags

5. The tab “Project” must contain information about Talent’s project that includes data about the duration of the project, composition and specialization of the team, the scope of the project, project preview

6. The layout must match the design posted on Figma (specified above)

### PR DESCRIPTION
**Pull request by illia-dobro**
This is UI with mockup data.
1s
2s
3s
4s

### CODE REVIEW

File: mobile/src/bundles/talent/components/badge/badge.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ type Properties = {
     value?: string | number;
     badgeType: BadgeName;
     iconSize?: number;
+    size?: string;
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/badge/badge.tsx]: we already have this on their PRs see how it's implemented there

File: mobile/src/bundles/talent/components/feedbacks/feedbacks.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import { Tag, Text, View } from '~/bundles/common/components/components';
+import { TextCategory } from '~/bundles/common/enums/enums';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/feedbacks/feedbacks.tsx]:
```suggestion
import { globalStyles } from '~/bundles/common/styles/styles';
```

File: mobile/src/bundles/talent/components/feedbacks/feedbacks.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import { styles } from './styles';
+
+type Properties = {
+    personalityType: string[];
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/feedbacks/feedbacks.tsx]:
```suggestion
    personalityTypes: string[];
```

File: mobile/src/bundles/talent/components/feedbacks/feedbacks.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    HRBadges: string[];
+};
+
+const Feedbacks = ({ personalityType, HRBadges }: Properties): JSX.Element => {
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/feedbacks/feedbacks.tsx]:
```suggestion
const FeedbacksContainer = ({ personalityType, HRBadges }: Properties): JSX.Element => {
```

File: mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    View,
+} from '~/bundles/common/components/components';
+import { TextCategory } from '~/bundles/common/enums/enums';
+import { globalStyles } from '~/bundles/common/styles/styles';
+import { type BsaBadgeStepBadgesTitle } from '~/bundles/talent/enums/enums';
+import { ProfileTab } from '~/bundles/talent/enums/enums';
+
+import { Feedbacks } from '../feedbacks/feedbacks';
+import { Project } from '../project/project';
+import { ScoresAndSkills } from '../scores-and-skills/scores-and-skills';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx]:
```suggestion
import { FeedbacksContainer } from '../feedbacks/feedbacks';
import { ProjectContainer } from '../project/project';
import { ScoresAndSkillsContainer } from '../scores-and-skills/scores-and-skills';
```

File: mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    hardSkills: string[];
+    personalType: string[];
+    HRBadges: string[];
+};
+
+const PreviewTabs = ({
+    badges,
+    hardSkills,
+    personalType,
+    HRBadges,
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx]: I guess you can select this data from the store, we already store steps values there

File: mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    const [tab, setTab] = useState<Tab>(ProfileTab.SCORES_SKILLS);
+    return (
+        <>
+            <View
+                style={[
+                    globalStyles.flexDirectionRow,
+                    globalStyles.justifyContentSpaceBetween,
+                ]}
+            >
+                {Object.values(ProfileTab).map((profileTab: Tab) => {
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx]: Move object values out of component

File: mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+                {tab === ProfileTab.SCORES_SKILLS && (
+                    <ScoresAndSkills badges={badges} skills={hardSkills} />
+                )}
+                {tab === ProfileTab.FEEDBACKS && (
+                    <Feedbacks
+                        personalityType={personalType}
+                        HRBadges={HRBadges}
+                    />
+                )}
+                {tab === ProfileTab.PROJECT && <Project />}
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx]: Switch case use memo

File: mobile/src/bundles/talent/components/profile-preview/profile-preview.tsx [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+import { Text, View } from '~/bundles/common/components/components';
+import { Color, IconName, TextCategory } from '~/bundles/common/enums/enums';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
+
+import { BsaBadgeStepBadgesTitle } from '../../enums/enums';
+import { PreviewTabs } from '../preview-tabs/preview-tabs';
+import { styles } from './styles';
+
+// TODO replace with real user data
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/profile-preview/profile-preview.tsx]: We need to get this data from the store

**Review by katerynakostikova — CHANGES_REQUESTED** [review]

File: mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx (hunk 6 above)
Comment (illia-dobro) [reply mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx]: HRBadges this is hard-coded data (in feedback)

I use props drilling from PorfilePreview, or useAppSelector here is better option?

Comment (katerynakostikova) [reply mobile/src/bundles/talent/components/preview-tabs/preview-tabs.tsx]: Let's use useAppSelector

File: mobile/src/bundles/talent/components/profile-preview/profile-preview.tsx [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+        'Problem-solving',
+        'Leadership',
+    ],
+};
+
+const iconSize = 24;
+type BadgeName = ValueOf<typeof BsaBadgeStepBadgesTitle>;
+
+const ProfilePreview: React.FC = () => {
+    const { onboardingData } = useAppSelector(({ talents }) => talents);
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/profile-preview/profile-preview.tsx:19]: Let's also destructure onboardingData to make code cleaner

File: mobile/src/bundles/talent/components/project-container/project-container.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+            {/* todo replace with actual data */}
+            <Text
+                category={TextCategory.CAPTION}
+                style={[styles.text, globalStyles.pb10]}
+            >
+                6 weeks / 6 engineers, 2 QA / JS / Healthtech industry
+            </Text>
+            <Image
+                // eslint-disable-next-line unicorn/prefer-module
+                source={require('~/assets/images/project.png')}
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/project-container/project-container.tsx]: fix this, see other's code

File: mobile/src/bundles/talent/components/project-container/project-container.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+import { Image, Text, View } from '~/bundles/common/components/components';
+import { TextCategory } from '~/bundles/common/enums/enums';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/project-container/project-container.tsx]:
```suggestion
import { globalStyles } from '~/bundles/common/styles/styles';
```

File: mobile/src/bundles/talent/components/scores-and-skills-container/scores-and-skills-container.tsx [hunk 13]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import {
+    type BsaBadgeStepBadgesTitle,
+    type SkillsStepDto,
+    type ValueOf,
+} from 'shared/build/index';
+
+import { Tag, Text, View } from '~/bundles/common/components/components';
+import { BadgeSize, TextCategory } from '~/bundles/common/enums/enums';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/components/scores-and-skills-container/scores-and-skills-container.tsx]:
```suggestion
import { globalStyles } from '~/bundles/common/styles/styles';
```

File: mobile/src/bundles/talent/screens/preview/preview.tsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+import { type TalentOnboardingScreenName } from '~/bundles/common/enums/enums';
+import {
+    ButtonType,
+    TalentOnboardingScreenNumber,
+} from '~/bundles/common/enums/enums';
+import { useAppRoute } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
+
+import { NewAccountHeader } from '../../components/components';
+import { ProfilePreview } from '../../components/profile-preview/profile-preview';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/screens/preview/preview.tsx]:
```suggestion
import { ProfilePreview } from '../../components/components';
```

Comment (katerynakostikova) [reply mobile/src/bundles/talent/screens/preview/preview.tsx]: and let's use absolute path, not ../../

File: mobile/src/bundles/talent/screens/preview/preview.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+    ScrollView,
+    View,
+} from '~/bundles/common/components/components';
+import { type TalentOnboardingScreenName } from '~/bundles/common/enums/enums';
+import {
+    ButtonType,
+    TalentOnboardingScreenNumber,
+} from '~/bundles/common/enums/enums';
+import { useAppRoute } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/screens/preview/preview.tsx]:
```suggestion
import { globalStyles } from '~/bundles/common/styles/styles';
```

File: mobile/src/bundles/talent/screens/preview/preview.tsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+            >
+                <ProfilePreview />
+
+                <View>
+                    <Button
+                        label="Save without publishing"
+                        buttonType={ButtonType.OUTLINE}
+                        style={globalStyles.mb10}
+                    />
+                    <Button label="Publish now" style={globalStyles.mb25} />
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/screens/preview/preview.tsx:51]: todo: add logic comment

**Review by katerynakostikova — CHANGES_REQUESTED** [review]

**Review by katerynakostikova — APPROVED** [review]

**Review by likeri29 — APPROVED** [review]

**Pull request merged**